src/jit/bitsetasshortlong.h

```diff
@@ -505,15 +504,12 @@ class BitSetOps</*BitSetType*/ BitSetShortLongRep,
                 if (hasBit)
                 {
                     *pElem = m_bitNum + nextBit;
-                    m_bitNum += nextBit + 1;
-                    m_bits >>= nextBit;
-                    m_bits >>= 1; // Have to do these separately -- if we have 0x80000000, nextBit == 31, and shifting
-                                  // by 32 bits does nothing.
+                    m_bits &= ~(((size_t)1) << nextBit); // clear bit we just found so we don't find it again
                     return true;
                 }
                 else
                 {
-                    unsigned len = BitSetTraits::GetArrSize(env, sizeof(size_t));
```

Comment:
For a further improvement you could hoist this call into the constructor and replace `m_env` with `m_len`.